### DISCUSSION TITLE
Global action filters with HttpEndPoint activity

### DISCUSSION THREAD

**Discussion by ricksearcy** [elsa-workflows/elsa-core#1231] [Q&A]
I have a need to use an action filter for all incoming requests to the workflow server. This works fine for any endpoints that are defined by Elsa and or my service, but it does not work for endpoints that are created by Elsa workflows as HttpEndPoint activities. Do you a recommended way to hook into the request pipeline when using an HttpEndpoint activity?

**Comment by sfmskywalker** [comment]
The HTTP Endpoint activities are invoked using the `HttpEndpointMiddleware` component. You can add your own middleware before this one.

For example, in `Startup`:

```csharp
public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
{
    app.UseCors();
    app.UseStaticFiles();
    app.UseRouting();
    app.UseAuthorization();
    app.UseMyOwnMiddleware(); // This would be your own middleware that executes before the next one.
    app.UseHttpActivities(); // This line installs the `HttpEndpointMiddleware` component.
    app.UseEndpoints(endpoints =>
    {
        endpoints.MapControllers();
    });
}
```

Let me know if this suffices. If not, we might consider implementing something similar to ASP.NET Core Action Filters for `HttpEndpointMiddleware`.

**Reply by ricksearcy** [reply]
Thanks! That will work for my use case.
